### PR TITLE
Watcher: fix metric stats names

### LINKED ISSUE #34865: Watcher Stats metrics parameters are mislabeled
While working testing HL REST client for Watcher Stats API I have noticed that the metirics parameters seem to be mislabeled:
```
GET /_xpack/watcher/stats/queued_watches
```
returns
```
{
...
      "current_watches": []
...
}
```
while 
```
GET /_xpack/watcher/stats/pending_watches
```
returns
```
{
...
      "queued_watches": []
...
}
```

After discussions with @spinscale and @martijnvg, @spinscale suggested that we should
1. `queued_watches` metric should return `queued_watches` instead of `current_watches` and
2. `pending_watches` should be renamed into `current_watches`

This is a breaking change, but since nobody noticed reported this discrepancy in the last 2 years, its impact will probably be relatively minimal.

### PR DESCRIPTION
**Pull request by imotov**
The current watcher stats metric names doesn't match the current
documentation. This commit fixes the behavior of `queued_watches`
metric, deprecates `pending_watches` metric and adds `current_watches`
to match the documented behavior. It also fixes the documentation, which
introduced `executing_watches` metric that was never added.

Fixes #34865

### CODE REVIEW

**Review by hub-cap — APPROVED** [review]
lol, looks like youve dealt with deprecations before :P 

LGTM but would like @spinscale to have a look since im not super familiar w these params

File: x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml [hunk 1]
```diff
@@ -12,3 +12,55 @@
         emit_stacktraces: "true"
   - match: { "manually_stopped": false }
   - match: { "stats.0.watcher_state": "started" }
+
+---
+"Test watcher stats current watches":
+  - skip:
+      version: " - 6.99.99"
+      reason:  materics were fixed in 7.0.0
```
Comment (spinscale) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml]: typo: `metrics`?

File: x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml [hunk 2]
```diff
@@ -12,3 +12,55 @@
...
+        metric: "current_watches"
+
+  - is_false: stats.0.queued_watches
+  - is_true: stats.0.current_watches
+
+---
+"Test watcher stats queued watches":
+  - skip:
+      version: " - 6.99.99"
+      reason:  materics were fixed in 7.0.0
```
Comment (spinscale) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml]: typo: `metrics`?

File: x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml [hunk 3]
```diff
@@ -12,3 +12,55 @@
...
+        metric: "queued_watches"
+
+  - is_false: stats.0.current_watches
+  - is_true: stats.0.queued_watches
+
+---
+"Test watcher stats queued watches using pending_watches":
+  - skip:
+      version: " - 6.99.99"
+      reason:  materics were fixed in 7.0.0
```
Comment (spinscale) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/watcher/stats/10_basic.yml]: typo: `metrics`?

**Review by spinscale — APPROVED** [review]
Minor typo. LGTM otherwise. Thanks for doing it!

**Pull request merged**